src/EventSubscriber/StrawberryEventSaveFileDeleteFlavorSubscriber.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Drupal\strawberryfield\EventSubscriber;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\file\FileInterface;
+use Drupal\search_api\Utility\Utility;
+use Drupal\strawberryfield\Event\StrawberryfieldCrudEvent;
+use Drupal\strawberryfield\Plugin\search_api\datasource\StrawberryfieldFlavorDatasource;
+use Drupal\strawberryfield\Tools\Ocfl\OcflHelper;
+
+/**
+ * Event subscriber for SBF bearing entity json process event.
+ */
+class StrawberryEventSaveFileDeleteFlavorSubscriber extends StrawberryfieldEventSaveSubscriber {
```

Comment:
There is an important piece missing here: the priority. We must be implicit that this needs to run as late as possible because some other even subscriber could affect which files are or not attached. Look at how that is done in the other subscribers and please test.

Reply:
Not sure I understand your comment. If we would do what you suggest with every event Subscriber we have and we put 9999 and hope for the best would that even work? Or said different, can you prove to me  that all works?

The rationale: Each ADO has different processing moments and some (the order) depends on others. File processing requires that files are organized and cleared from the as:structure, etc, some run initially, others later, others as late as we can.
9999 means also `The higher the number, the earlier the method is called`

So why ask here is for you to check if your even subscriber may need data/processing or a state that some other event subscriber in our archipelago implementation (Strawberryfield and Strawberry Runners) processed before yours (for the same event). Or you want to be the first one. Just that. Conclussion may be that 100 (default for this base class) is right. But I do not want the hope for the best approach

Reply:
I'm not explaining what I meant very well, apologies for that.
A priority is a number, it could be 100, 90210 or -1.9 trillion, what I'm trying to say is that this reflects certain arbitrarity, it is not a fully fledged dependency system. Meaning, I could add whatever number and we have no way to control if it's the right one. If we anticipate this is an event with many clients, we might as well *name* the main events with a constant, a final faking an enum or whatever, so instead of saying "100" we say  "after_file_is_processed" or just "default" if we don't really care. At the end of the day the control we're going to have on these events is very slim anyways (a third party could subscribe when we arrive to the point that there are contribs outside our core ecosystem)

Anyway, these are the ones that I've seen subscribed so far:

```
WebformStrawberryfieldDeleteTmpStorage, priority -700. Event subscriber that deletes Webform temp storage for SBF bearing entities.
formatStrawberryfieldDeleteTmpStorage, priority -700. Event subscriber that deletes format temp storage for SBF bearing entities.

Dependant on StrawberryfieldEventSaveSubscriber parent:

StrawberryRunnersEventSavePostProcessingSubscriber, priority -2000. Event subscriber for SBF bearing entity when a file is deleted.
```

The only one that could affect is the latter, but I don't think is dealing with the same stuff, so unless there's a reason that I'm not seeing for changing the priority, I'd suggest to leave it as it is (I've tested this a lot looking at the queues), but again, I might be horribly wrong based on your feedback.

Reply:
Ok. If you tested this and **100** (default inherited from the base class) is OK please make it **local** (means fix it to 100 in your class) to your event subscriber. My real concern other than the order of execution (not the dependencies, there are none) is if we decide (ever) to change the base class default I'm pretty sure we won't remember this. 
In this case `100`means removal of old SBR documents in Solr (`StrawberryRunnersEventSavePostProcessingSubscriber`) will run before enqueuing of new ones. There, We decide if we enqueue (for Solr based outputs) based on if the Solr DOC ID to be processed is already in Solr. Since yours untracks but really does not ensure the removal of index happens (and that is fine) it is OK to have `StrawberryRunnersEventSavePostProcessingSubscriber` as later as possible and yours before. 

I'm happy with slim control. I do care about the order of operations, more on the PreSave Subscriber of course than here. (e.g some require that the AS:STRUCTURE is in place already)